curve25519-dalek/src/backend/serial/u64/common_verus/div_mod_lemmas.rs

```diff
@@ -91,6 +90,338 @@ pub proof fn lemma_div_of_sum(a: nat, b: nat, k: nat)
 
     lemma_div_multiples_vanish_fancy((a0 + b0) as int, ((a % k) + (b % k)) as int, k as int);
 }
+/// Lemma: Uniqueness of quotient in division
+/// If value = q * d + r with 0 <= r < d, then value / d = q
+///
+/// This establishes that the quotient in the fundamental div/mod theorem is unique.
+pub proof fn lemma_div_quotient_unique(value: int, d: int, q: int, r: int)
+    requires
+        d > 0,
+        value == q * d + r,
+        0 <= r < d,
+    ensures
+        value / d == q,
+{
+    // Use fundamental division theorem
+    lemma_fundamental_div_mod(value, d);
+    lemma_mod_bound(value, d);
+
+    let q_actual = value / d;
+    let r_actual = value % d;
+
+    // From fundamental theorem, we get the relationship (but stated with multiplication on the left)
+
+    // Convert to multiplication on the right for easier comparison
+    assert(d * q_actual == q_actual * d) by { lemma_mul_is_commutative(d, q_actual); }
+    assert(d * q == q * d) by { lemma_mul_is_commutative(d, q); }
+    assert(d * (q - q_actual) == r_actual - r) by {
+        lemma_mul_is_distributive_sub(d, q, q_actual);
+    }
+
+    // Key insight: Since both r and r_actual are in [0, d), their difference is in (-d, d)
+    // But d * (q - q_actual) must be a multiple of d
+    // The only multiple of d in the range (-d, d) is 0
+    // Therefore: d * (q - q_actual) = 0, which means q = q_actual
+
+    if q != q_actual {
+        // If q != q_actual, then |q - q_actual| >= 1
+        let diff = q - q_actual;
+
+        // Therefore |d * (q - q_actual)| >= d
+        if diff > 0 {
+            lemma_mul_left_inequality(d, 1, diff);
+        } else {
+            lemma_mul_inequality(diff, -1, d);
+        }
+    }
+
+}
+
+/// Lemma: For x < d, x % d = x
+/// This is a fundamental property of modulo for values less than the divisor.
+pub proof fn lemma_mod_of_less_than_divisor(x: int, d: int)
+    requires
+        d > 0,
+        0 <= x < d,
+    ensures
+        x % d == x,
+{
```

Comment:
This is just `lemma_small_mod`

https://verus-lang.github.io/verus/verusdoc/vstd/arithmetic/div_mod/fn.lemma_small_mod.html